prejpms/src/main/java/com/baeldung/prejpms/App.java

```diff
@@ -0,0 +1,79 @@
+package com.baeldung.prejpms;
+
+import java.io.StringWriter;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Marshaller;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.sun.crypto.provider.SunJCE;
+
+import sun.misc.BASE64Encoder;
+import sun.reflect.Reflection;
+
+public class App {
+
+    private static final Logger logger = LoggerFactory.getLogger(App.class);
+
+    public static void main(String[] args) throws Exception {
+
+        getCrytpographyProviderName();
+        getCallStackClassNames();
+        getXmlFromObject(new Book(100, "Java Modules Architecture"));
+        getBase64EncodedString("Java");
+    }
+
+    private static void getCrytpographyProviderName() {
+        try {
+            logger.info("1. Java Cryptography Extension - Provider Name: " + new SunJCE().getName() + "\n");
+        } catch (Throwable e) {
+            logger.error(e.toString());
+        }
+    }
+
+    private static void getCallStackClassNames() {
+        try {
+            int i = 0;
+            StringBuffer sbStack = new StringBuffer();
+            while (true) {
+                Class<?> caller = Reflection.getCallerClass(i++);
+                if (caller == null) {
+                    break;
+                } else {
+                    sbStack.append(caller.getName())
+                        .append("\n");
+                }
+            }
+            logger.info("2. Call Stack Class Names:\n" + sbStack.toString());
```

Comment:
Does this logger allow interpolation via `{}` ? If so, it may make the examples easier to read, especially in the version of logging statements pasted into the article.

E.g.

```
// also, try to keep the text shorter, with the objective
// of making it the least to read in the wordpress post - think mobile
logger.info("2. Call Stack:\n {}", sbStack);
```